validphys2/src/validphys/filters.py

```diff
@@ -645,7 +645,12 @@ def __call__(self, dataset, idat):
         # is different to the case where the rule does apply,
         # but the point was cut out by the rule.
         if (
-            (dataset.setname != self.dataset and dataset.legacy_name != self.dataset)
+            not (
+                dataset.setname == self.dataset
+                or
+                # for old rules, we might be using an old name of this dataset
+                (dataset.legacy_names is not None and self.dataset in dataset.legacy_names)
+            )
```

Comment:
Wouldn't it better to split these into multiple if statements? Having so many conditions in one single statement is a bit too much and (surely) breaks some PEP8 guides.